### PR TITLE
post, interface: The Go Interface

### PR DESCRIPTION
**Pull request by EdDev**

### CODE REVIEW

File: _posts/2021-01-20-the-go-interface.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    It("succeed write to store", func() {
+        store := testStore{}
+        Expect(app.Save([]byte{}, store)).To(Succeed())
+    })
+})
+```
+
+## Please avoid these
+
+### Return an interface
```
Comment (RamLavi) [inline _posts/2021-01-20-the-go-interface.md:283]: can you add a simple example?

File: _posts/2021-01-20-the-go-interface.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+behavior which is used by its consumers.
+But this limits the concrete type to expand and support other behaviors (it is
+limited by the interface).
+
+It may also be a premature abstraction.
+
+But there are always exceptions. A factory that may return different concrete
+types must have an interface as the return value.
+
+### Define interfaces side by side with concrete objects
```
Comment (RamLavi) [inline _posts/2021-01-20-the-go-interface.md:335]: ditto

**Review by RamLavi — CHANGES_REQUESTED** [review]
small request to add examples for the avoid points :)

File: _posts/2021-01-20-the-go-interface.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+
+Here is a simple interface:
+```golang
+type Runner interface {
+    Run(distance int) error
+}
+```
+
+Any concrete implementation that implements the `Run` function signature
+can be place "under" the `Runner` interface.
```
Comment (RamLavi) [inline _posts/2021-01-20-the-go-interface.md]:
```suggestion
can be placed "under" the `Runner` interface.
```

**Review by RamLavi — CHANGES_REQUESTED** [review]
nit

File: _posts/2021-01-20-the-go-interface.md (hunk 3 above)
Comment (EdDev) [reply _posts/2021-01-20-the-go-interface.md]: Done

File: _posts/2021-01-20-the-go-interface.md (hunk 1 above)
Comment (EdDev) [reply _posts/2021-01-20-the-go-interface.md:283]: Done

File: _posts/2021-01-20-the-go-interface.md (hunk 2 above)
Comment (EdDev) [reply _posts/2021-01-20-the-go-interface.md:335]: I can ref an example (`io` package). Expressing this in code using different packages will add a lot of code.

Comment (RamLavi) [reply _posts/2021-01-20-the-go-interface.md:335]: I agree. If you can refer to an already existing example then it would be best.

**Review by RamLavi — APPROVED** [review]
/lgtm

**Pull request merged**